qiskit/pulse/instructions/play.py

```diff
@@ -31,31 +33,98 @@ class Play(Instruction):
     cycle time, dt, of the backend.
     """
 
-    def __init__(self, pulse: Pulse, channel: PulseChannel, name: Optional[str] = None):
+    def __init__(
+        self,
+        pulse: Pulse,
+        mixed_frame: Optional[Union[PulseChannel, MixedFrame]] = None,
+        pulse_target: Optional[PulseTarget] = None,
+        frame: Optional[Frame] = None,
+        name: Optional[str] = None,
+        channel: Optional[PulseChannel] = None,
+    ):
```

Comment:
```suggestion
        self,
        pulse: Pulse,
        mixed_frame: Optional[Union[PulseChannel, MixedFrame]] = None,
        *,
        pulse_target: Optional[PulseTarget] = None,
        frame: Optional[Frame] = None,
        name: Optional[str] = None,
        channel: Optional[PulseChannel] = None,
    ):
```
since we have too much flexibility in defining the channel now, I think asking user to provide keyword is safer.

Reply:
I like the general idea, but I don't think it's beneficial to have `mixed_frame` as a positional argument. What if I want to specify the instruction with `frame` and `pulse_target`? I'll have to use `Play(None, frame=..., pulse_target=...)`.

The catch is, we can't set all arguments as forced keywords (for backwards compatibility), and we can't choose one to be positional.